frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { k8sList, K8sResourceKind } from '@console/internal/module/k8s';
+import { ClusterTaskModel, TaskModel } from '../../../models';
+import { PipelineTask } from '../../../utils/pipeline-augment';
+import { PipelineVisualizationTaskItem } from '../../../utils/pipeline-utils';
+import { PipelineMixedNodeModel, PipelineTaskListNode } from '../pipeline-topology/types';
+import {
+  createTaskListNode,
+  handleParallelToParallelNodes,
+  tasksToBuilderNodes,
+} from '../pipeline-topology/utils';
+import { AddNodeDirection } from '../pipeline-topology/const';
+import { convertResourceToTask } from './utils';
+
+type UseTasks = {
+  namespacedTasks: K8sResourceKind[] | null;
```

Comment:
we dont need | null, do we?

Reply:
No, I suppose that is true... I was just trying to the actual values. I guess any object can be null.